carbon-design-system/carbon
# fix(toggle): ensure onToggle still work with props['aria-labelledby']

Closes #13004

#### Changelog

**New**

- If `props['aria-labelledby']` is passed the wrapper element listens for a `click` event to trigger `handleClick` as the labeling element is no longer a `<label>` but a `<div>` and the `<button>` is visually hidden.

**Changed**

- Clarified the relationship between `props.hideLabel` and `props.labelText` in the prop description of `props.hideLabel`

#### Testing / Reviewing

- In the storybook, add a dummy `props.onClick` and `props.onToggle` function and verify that they are triggered both by clicking on the toggle and by using the keyboard **when `props['aria-labelledby']` is passed**.
  - Verify that these functions are only called once per click/keyboard interaction.
  - Verify that these functions are also called when `props['aria-labelledby']` is not passed.
